service/app/distribution.go

```diff
@@ -58,11 +58,12 @@ type Pack struct {
 	DistributionID uuid.UUID
 	ID             uuid.UUID `gorm:"column:id;primary_key;type:uuid;"`
 
-	FlowID         common.FlowID      `gorm:"column:flow_id;index"`         // ID of the pack NFT
-	State          common.PackState   `gorm:"column:state"`                 // public
-	Salt           common.BinaryValue `gorm:"column:salt"`                  // private
-	CommitmentHash common.BinaryValue `gorm:"column:commitment_hash;index"` // public
-	Collectibles   Collectibles       `gorm:"column:collectibles"`          // private
+	ContractReference AddressLocation    `gorm:"embedded;embeddedPrefix:contract_ref_"` // Reference to the pack NFT contract
+	FlowID            common.FlowID      `gorm:"column:flow_id;index"`                  // ID of the pack NFT
```

Comment:
Should there be a unique (multi-column) restraint for `ContractReference` & `FlowID`?

Reply:
That would represent the correct data layout _if_ we would have the FlowID when creating a Pack in the database. But we get the FlowID from chain later on when minting the packs so we will have duplicates until that.

Reply:
Ok, makes sense 👍